metadata.json

```diff
@@ -19,6 +19,10 @@
     {
       "name": "puppetlabs/concat",
       "version_requirement": ">= 1.1.0 < 7.0.0"
+    },
+    {
+      "name": "puppet/selinux",
```

Comment:
hey people. I'm wondering if the selinux module should stay a soft dependency because it's just used at one place in this module.

Reply:
@bastelfreak selinux is a pretty critical part of the OS and if this module needs to modify SELinux settings I would say it is not a soft dependency.

Reply:
but it's only a critical part if it's a box with selinux where selinux is actually enabled and not handled already in another profile/module. I assume that's a valid statement for only a small number of users.

Reply:
I understand but selinux is enabled by default in both RHEL and Fedora which covers a lot of use cases.  How would I mark a module as a soft dependency?  There doesn't seem to be an option for that in puppet metadata files.

Reply:
So, my understanding is that there is a case in which this module will error if the selinux module is not present, and that is where:

- the target node has selinux enabled
- selinux is not being managed by another profile or module
- the node is being managed with this module

Puppet does not distinguish between hard/soft deps, a module is either required or not in the metadata. One option here would be to document the edge case if you don't want to actually include the module in the metadata dependencies.

Reply:
The module will error out if selinux is enabled and the postgresql alternate port is not labeled properly before the service is started.  The default SELinux policies only allow postgresql to bind to port 5432.

I originally added selinux as a dependency in the metadata file but removed it as I was told we don't want to make it a hard requirement.

IMO the selinux is almost as essential as stdlib.  SELinux is enabled by default on Redhat, Fedora, and I believe OracleLinux since that is based on RHEL but I could be mistaken.

Reply:
I've done some additional checking and it looks like the puppet-selinux module does not support RedHat 5 so I will need to revisit how to manage SELinux ports.  I will updated this PR once I have a solution.

Reply:
I've updated the manifest to use an exec resource instead of the selinux::port type.  This eliminates the dependency on the selinux module and will work with all platforms supported by puppet.